BaseStation/MainWindow/Settings/SettingsPanel.xaml.cs

```diff
@@ -72,9 +72,25 @@ private void initMapFiles()
 
         private void Button_GetMap(object sender, RoutedEventArgs e)
         {
-            MapTileDownloadManager.DownloadNewTileSet(MapConfig);
-            initMapFiles();
-            Settings.CurrentMapFile = MapConfig.MapSetName + ".map";
+            MapStatus.Content = "Downloading map...";
```

Comment:
I was thinking about doing this but you implemented it way better than i would have. What I would have done is have the button be disabled while the thread is running so you cant spam it.